Raven.Tests.Helpers/RavenTestBase.cs

```diff
@@ -865,5 +962,65 @@ protected static void DeployNorthwind(DocumentStore store, string databaseName =
 			var request = requestFactory.CreateHttpJsonRequest(new CreateHttpJsonRequestParams(null, url + "/studio-tasks/createSampleData", "POST", store.DatabaseCommands.PrimaryCredentials, store.Conventions));
 			request.ExecuteRequest();
 		}
+
+        private static void StoreSeedData(IEnumerable<IEnumerable> seedData,
```

Comment:
It is better to have multiple overloads, rather than something that does so
much dynamic magic.

*Hibernating Rhinos Ltd  *

Oren Eini* l CEO l *Mobile: + 972-52-548-6969

Office: +972-4-622-7811 *l *Fax: +972-153-4-622-7811



On Wed, May 20, 2015 at 12:28 PM, Pure Krome <notifications@github.com>
wrote:

> In Raven.Tests.Helpers/RavenTestBase.cs
> <https://github.com/ravendb/ravendb/pull/592#discussion_r30684206>:
>
> > @@ -865,5 +962,65 @@ protected static void DeployNorthwind(DocumentStore store, string databaseName =
> >  			var request = requestFactory.CreateHttpJsonRequest(new CreateHttpJsonRequestParams(null, url + "/studio-tasks/createSampleData", "POST", store.DatabaseCommands.PrimaryCredentials, store.Conventions));
> >  			request.ExecuteRequest();
> >  		}
> > +
> > +        private static void StoreSeedData(IEnumerable<IEnumerable> seedData,
>
> This method signature is interesting. Right now it's designed to take a
> collection of collections:
>
> eg.
>
> var seedData = new IEnumerable[]
>         {
>             FakeModelHelpers.CreateFakeCats(),
>             FakeModelHelpers.CreateFakeDogs()
>         };
>
> But i was wondering if it might be kewl to make it accept just a single
> object and I try and see if that object is Enumerable and if so, iterate
> through it? (and do that 2x - first for the top collection, 2nd check for
> the items in each collection).
>
> Why? just to make it easier for a developer to consume this method.
> eg.
>
> var seedData = new Cat();
> var seedData = CreateFakeCats(); // IEnumerable<Cat>
> var seedData = new IEnumerable[] { .. // as above in the top of this comment ... };
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ravendb/ravendb/pull/592/files#r30684206>.
>